fw/http_parser.c

```diff
@@ -5915,8 +5960,27 @@ Req_Method_1CharStep: __attribute__((cold))
 
 	/* Host port in URI */
 	__FSM_STATE(Req_UriPort, cold) {
-		if (likely(isdigit(c)))
-			__FSM_MOVE_f(Req_UriPort, &req->host);
+		__fsm_sz = __data_remain(p);
+		__fsm_n = __parse_uri_port(p, __data_remain(p),
+					   &parser->_acc, USHRT_MAX);
+
+		switch (__fsm_n) {
+		case CSTR_BADLEN:
+		case CSTR_NEQ:
+			TFW_PARSER_DROP(Req_UriPort);
+		case CSTR_POSTPONE:
+			req->host_port = parser->_acc;
+			__FSM_MOVE_nf(Req_UriPort, __fsm_sz, &req->host);
```

Comment:
In `Req_UriAuthorityEnd` we use __FSM_MOVE_f(Req_UriPort, &req->host); but here __FSM_MOVE_nf why?

Reply:
__FSM_MOVE_f it's `__FSM_MOVE_nf` with 1 byte step.
Definition:
```  #define __FSM_MOVE_f(to, field)		__FSM_MOVE_nf(to, 1, field) ```